### PR TITLE
HDDS-10680. Duplicate delete key blocks sent to SCM

### PR DESCRIPTION
**Pull request by ChenSammi**
## What changes were proposed in this pull request?

Fix the condition that causes duplicate command sent to SCM. 
Please find the detail analysis in JIRA description. The issue caused SCM frequent crash in user's environment. 

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-10680

## How was this patch tested?

Verified in user's cluster.

### CODE REVIEW

**Review by sumitagrawl — APPROVED** [review]
LGTM

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/AbstractKeyDeletingService.java
Comment (smengcl) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/AbstractKeyDeletingService.java:1]: Is the full patch uploaded? I do not see much changes here other than the logging additions?

Comment (ChenSammi) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/AbstractKeyDeletingService.java:1]: The major change is "Preconditions.assertTrue" condition change. There is one "/volume/bucket/" in the deletedTable which breaks the assertion, which cause the deletedBlocks sent to SCM already not deleted from OM's RocksDB, which in turn cause the duplicate blocks sent to SCM again and again. It has found that same blockID is in SCM's audit log for hundreds of times.

**Comment by ChenSammi** [comment]
Thanks @sumitagrawl and @smengcl for review.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/AbstractKeyDeletingService.java
Comment (smengcl) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/AbstractKeyDeletingService.java:1]: Thanks @ChenSammi for the clarification.

**Pull request merged**